task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -361,6 +363,29 @@ def get_relevant_upstream_map_indexes(
         # TODO: Implement this method
         return None
 
+    def get_first_reschedule_date(self, try_number: int) -> datetime | None:
```

Comment:
When is this called?

Reply:
To get duration for `timeouts` and to get `next_poll_interval`

https://github.com/apache/airflow/blob/b8b6c594ba3fbd03b24c6e8df6c708c2901041d2/airflow-core/src/airflow/sensors/base.py#L205-L213